snuba/snuba_migrations/profile_chunks/0001_create_profile_chunks_table.py

```diff
@@ -0,0 +1,83 @@
+from typing import List, Sequence
+
+from snuba.clickhouse.columns import UUID, Column, DateTime64, UInt
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations, table_engines
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+from snuba.migrations.operations import OperationTarget, SqlOperation
+
+storage_set = StorageSetKey.PROFILES
+table_prefix = "profile_chunks"
+local_table_name = f"{table_prefix}_local"
+dist_table_name = f"{table_prefix}_dist"
+
+columns: List[Column[Modifiers]] = [
+    Column("project_id", UInt(64)),
+    Column("profiler_id", UUID()),
+    Column("chunk_id", UUID()),
+    Column(
+        "start_timestamp",
+        DateTime64(
+            precision=6,
+            modifiers=Modifiers(codecs=["DoubleDelta"]),
+        ),
+    ),
+    Column(
+        "end_timestamp",
+        DateTime64(
+            precision=6,
+            modifiers=Modifiers(codecs=["DoubleDelta"]),
+        ),
+    ),
+    Column("retention_days", UInt(16)),
+    Column("partition", UInt(16)),
+    Column("offset", UInt(64)),
+    Column("deleted", UInt(8)),
+]
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    blocking = False
+
+    def forwards_ops(self) -> Sequence[SqlOperation]:
+        return [
+            operations.CreateTable(
+                storage_set=storage_set,
+                table_name=local_table_name,
+                columns=columns,
+                engine=table_engines.ReplacingMergeTree(
+                    order_by="(project_id, profiler_id, start_timestamp, end_timestamp, cityHash64(chunk_id))",
+                    partition_by="(retention_days, toStartOfDay(start_timestamp))",
+                    sample_by="cityHash64(chunk_id)",
+                    settings={"index_granularity": "8192"},
+                    storage_set=storage_set,
+                    ttl="toDateTime(end_timestamp) + toIntervalDay(retention_days)",
+                    version_column="deleted",
```

Comment:
This is a follow-up to the previous [comment](https://github.com/getsentry/snuba/pull/5895/files#r1599004882) and maybe someone from @getsentry/sns can chime-in and answer this.

I'm looking at the clickhouse docs for `ReplacingMergeTree` and a couple of things caught my attention:

- `is_deleted` can only be enabled when `ver` is used
- `ENGINE = ReplacingMergeTree(eventTime, is_deleted)`

from what I can see here though: 
1) we're using the _deleted_ column as the _version_column_, and I don't think that's what we want? 
2) we're only setting the `VER` (version column), not the `is_deleted` since the current Snuba definition of the ReplacingMergeTree seems to neither accept nor set this column (see [here](https://github.com/getsentry/snuba/blob/master/snuba/migrations/table_engines.py#L126)).

Reply:
This is correct with ClickHouse version >= 23.2.1 (https://github.com/ClickHouse/ClickHouse/commit/6526c2a8abbab6f027672903cf05e1537274f109). Before that, only the `ver` setting existed and I believe we used it like the `is_delete`. I believe the profiling cluster is still on 22.8.

Anyway, since then, they discovered the feature wasn't working well (https://github.com/ClickHouse/ClickHouse/issues/50346), removed it (https://github.com/ClickHouse/ClickHouse/pull/57932) and replaced it with (https://github.com/ClickHouse/ClickHouse/issues/54880).